sql/src/main/java/io/crate/metadata/cluster/AlterTableClusterStateExecutor.java

```diff
@@ -0,0 +1,437 @@
+        }
+
+        final IndexTemplateMetaData.Builder templateBuilder = IndexTemplateMetaData.builder(templateName)
+            .order(indexTemplateMetaData.order())
+            .patterns(indexTemplateMetaData.getPatterns())
+            .settings(settings)
+            .putMapping(Constants.DEFAULT_MAPPING_TYPE, mergeTemplateMapping(indexTemplateMetaData, alterTableRequest.sourceAsMap()));
+
+        for (ObjectObjectCursor<String, AliasMetaData> container : indexTemplateMetaData.aliases()) {
+            templateBuilder.putAlias(container.value);
+        }
+
+        final MetaData.Builder metaData = MetaData.builder(currentState.metaData()).put(templateBuilder.build());
+        return ClusterState.builder(currentState).metaData(metaData).build();
+    }
+
+    private Map<String, Object> parseMapping(String mappingSource) {
+        try (XContentParser parser = JsonXContent.jsonXContent
+            .createParser(NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION, mappingSource)) {
+            return parser.map();
+        } catch (IOException e) {
+            throw new ElasticsearchException("failed to parse mapping");
+        }
+    }
+
+    private String mergeTemplateMapping(IndexTemplateMetaData templateMetaData, Map<String, Object> newMapping) {
+        Map<String, Object> mergedMapping = new HashMap<>();
+        for (ObjectObjectCursor<String, CompressedXContent> cursor : templateMetaData.mappings()) {
+            Map<String, Object> mapping = parseMapping(cursor.value.toString());
+            Object o = mapping.get(Constants.DEFAULT_MAPPING_TYPE);
+            assert o instanceof Map : "o must not be null and must be instance of Map";
+            XContentHelper.update(mergedMapping, (Map) o, false);
+        }
+        XContentHelper.update(mergedMapping, newMapping, false);
+
+        try {
+            return Strings.toString(XContentFactory.contentBuilder(XContentType.JSON).map(MapBuilder.<String, Object>newMapBuilder().put(
+                Constants.DEFAULT_MAPPING_TYPE,
+                mergedMapping).map()));
+        } catch (IOException e) {
+            throw new ElasticsearchException("failed to serialize mapping");
+        }
+    }
+
+    private List<String> validateSettings(Settings settings) {
+        List<String> validationErrors = new ArrayList<>();
+        try {
+            indexScopedSettings.validate(settings, true); // templates must be consistent with regards to dependencies
+        } catch (IllegalArgumentException iae) {
+            validationErrors.add(iae.getMessage());
+            for (Throwable t : iae.getSuppressed()) {
+                validationErrors.add(t.getMessage());
+            }
+        }
+        List<String> indexSettingsValidation = metaDataCreateIndexService.getIndexSettingsValidationErrors(settings, true);
+        validationErrors.addAll(indexSettingsValidation);
+        return validationErrors;
```

Comment:
We could throw an exception directly here instead of returning a list of errors which will be converted into an exception